clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2230,6 +2225,193 @@ class SyclKernelArgsSizeChecker : public SyclKernelFieldHandler {
   using SyclKernelFieldHandler::handleSyclHalfType;
 };
 
+enum class KernelArgDescription {
```

Comment:
Would you know where to? I'm not very familiar with Sema-CG intefaces.

Reply:
There's two choices: AST or Basic. Target-specific stuff usually goes into Basic, and I think this functionality is sort diagnostic related, so I'd say Basic\Diagnostic.h or somewhere near there.

Reply:
So I was working on this when I wondered if it would be better if I just construct the required strings in Sema, (i.e. instead of passing ArgName, ArgDesc, etc and then constructing the output in CodeGen based on these values) I could just pass the string I want emitted.  I think it would get rid of a lot of unnecessary code

Reply:
If I'm following along properly, that sounds like a reasonable idea to me.

Reply:
Done